### DISCUSSION TITLE
Proposal to move IE11 support to an external module, remove jimp

### DISCUSSION THREAD

**Discussion by boutell** [apostrophecms/apostrophe#3812] [Announcements]
Now that Microsoft no longer supports Internet Explorer 11, there is no reasonable expectation that users should still be using it.

Yet the inclusion of the needed dependencies for our `es5: true` (IE11) feature **doubles the `npm install` time of Apostrophe,** which can be long on a slower computer.

This is because it requires `babel`, which is a very heavy dependency adding a lot of build time.

The case we're making here is that removing IE11 support is not a bc break because Microsoft itself has removed IE11 from the desktop and is pushing Edge upgrades on supported versions of Windows. Also removing support will not break existing projects that set `es5: true`, they just won't build IE11 bundles anymore unless an optional package is added.

So we're proposing to move this feature to an optional package, `@apostrophecms/ie11`, which will be maintained for as long as the community remains interested in doing so. Projects requiring `es5: true` would add this optional package to their dependencies to continue IE11 support.

In a related change, we would remove `jimp`, which is responsible for about a third of the remaining build time, in favor of `sharp`, which performs much better and faster image conversion using `libvips`, a high speed image manipulation library that is installed automatically by the `sharp` package and works across supported Windows, Linux and MacOS platforms.

Please let us know your thoughts on this plan. Thanks!

**Comment by ytilis** [comment]
I'm absolutely in favor of this. Pulling IE11 support out into a separate optional module makes sense given Apostrophe's modular nature and where IE11 is in its lifecycle and marketshare. Doubly so given the very real and tangible benefit it would provide to project which do not require IE11 support.
